File: solutions/julia/cars-assemble/12/cars-assemble.jl
const base_prod_rate = 221

function success_rate(speed)
    return if speed ≤ 0
        0.00
    elseif speed ≤ 4
        1.00
    elseif speed ≤ 8
        0.90
    elseif speed == 9
        0.80
    else
        0.77
    end
end

production_rate_per_hour(speed) = base_prod_rate * speed * success_rate(speed)

working_items_per_minute(speed) = Int(production_rate_per_hour(speed) ÷ 60)

File: solutions/java/cars-assemble/5/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private static final int BASE_PRODUCTION_RATE = 221;
    private static final int MINUTES_PER_HOUR = 60;

    public double productionRatePerHour(int speed) {
        return speed * this.actualProductionRate(speed);
    }

    public int workingItemsPerMinute(int speed) {
        return (int) this.productionRatePerHour(speed) / MINUTES_PER_HOUR;
    }

    private double actualProductionRate(final int speed) {
        if (speed <= 0) return 0;
        if (speed < 5)  return BASE_PRODUCTION_RATE;
        if (speed < 9)  return BASE_PRODUCTION_RATE * 0.9;
        if (speed == 9) return BASE_PRODUCTION_RATE * 0.8;
        return BASE_PRODUCTION_RATE * 0.77;
    }
}
